### PR TITLE
Harry fiftyfifty - updated ToggleButtonGroups into standalone components

### PR DESCRIPTION
**Pull request by hphilpotts**
PR heading says it all.

One query re. whether to import `useGuesses()` into both parent `FiftyFiftyRound` and respective child `ToggleGroup` components, or to just do this in the parent and pass guesses down as a prop...or something. Long day.

### CODE REVIEW

File: src/components/ToggleGroup.jsx [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+import { ToggleButton, ToggleButtonGroup } from '@mui/material'
+import { useGuesses } from '../context/GuessContext' // see line 11
+
+
+export const ToggleGroup = ({ values, labels, handleChange }) => {
+
+    const { name, value1, value2, value3 } = values
+    const { groupLabel, buttonLabel1, buttonLabel2, buttonLabel3 } = labels
+
+    const { guesses } = useGuesses() // ? pass as prop from parent [also has useGuesses()] instead?
```
Comment (mdnrosen) [inline src/components/ToggleGroup.jsx]: In this case its better to useContext as we're already passing quite a few props. Plus the whole point of our context here is to minimise passing the same props all the time

File: src/components/ToggleGroup.jsx [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    return (
+        <ToggleButtonGroup
+            fullWidth
+            value={guesses[name]}
+            exclusive
+            color="warning"
+            onChange={handleChange}
+            aria-label={groupLabel}
+        >
+            <ToggleButton name={name} value={value1} aria-label={buttonLabel1}>{value1}</ToggleButton>
```
Comment (mdnrosen) [inline src/components/ToggleGroup.jsx]: You have 3 buttons so I would use an array, then you can map over it. Pass the array of 'options' as props then use the index to pick out the data you need

File: src/pages/FiftyFiftyRound.jsx [hunk 3]
```diff
@@ -36,17 +37,11 @@ export const FiftyFiftyRound = () => {
...
-                                        exclusive
-                                        color="warning"
-                                        onChange={handleChange}
-                                        aria-label='true or false'
-                                    >
-                                        <ToggleButton name="h2h_superover" value="true" aria-label='true'>true</ToggleButton>
-                                        <ToggleButton name="h2h_superover" value="false" aria-label='false'>false</ToggleButton>
-                                    </ToggleButtonGroup>
+                                    <ToggleGroup
+                                        values={{ name: "h2h_superover", value1: "true", value2: "false" }}
```
Comment (mdnrosen) [inline src/pages/FiftyFiftyRound.jsx]: I think (as above) you can build an array for this data

const options = [{
   label: 'Pakistan',
   value: 'pakistan'
   ariaText: 'some text here'
}}
<ToggleGroup
     options={options} />

**Review by mdnrosen — CHANGES_REQUESTED** [review]

File: src/pages/FiftyFiftyRound.jsx (hunk 3 above)
Comment (mdnrosen) [reply src/pages/FiftyFiftyRound.jsx]: then pass that as props

**Comment by hphilpotts** [comment]
Reckon this should be good to go. 

Props for TG component passed as:

```
<ToggleGroup
  name=string // provides the key for guesses context
  values=[string1, string2, *string3] // provides the value for guesses context, TButton text, rendered using .map()
  labels=[string1, string2, string3, *string4] // provides aria-labels for TBGroup, TButtons 1, 2 and (optional) 3
  handleChange=function // ...
/>
```

Redundant imports, comments and `console.log` statements cleaned out.

**Review by mdnrosen — APPROVED** [review]

**Pull request merged**